File: ci360-react-native-ios-app/sdk/latest/example/src/models/JsonCreative.ts
class JsonCreative {
  title: string;
  subTitle: string;
  bullets: string[];
  buttonText: string;

  constructor(json: string | null) {
    try {
      const data = JSON.parse(json);
      if (!data) {
        this.title = '';
        this.subTitle = '';
        this.bullets = [];
        this.buttonText = '';
      } else {
        this.title = data.title;
        this.subTitle = data.subTitle;
        this.bullets = data.bullets || [];
        this.buttonText = data.buttonText;
      }
    } catch (error) {
      console.log('JSON parse error: ', error);
      this.title = '';
      this.subTitle = '';
      this.bullets = [];
      this.buttonText = '';
    }
  }
}

export default JsonCreative;

File: ci360-flutter-ios-app/sdk/latest/example/lib/models/json_creative.dart
class JsonCreative {
  final String title;
  final String subTitle;
  final List<String> bullets;
  final String buttonText;

  JsonCreative(
      {required this.title,
      required this.subTitle,
      required this.bullets,
      required this.buttonText});

  factory JsonCreative.fromJson(Map<String, dynamic> json) {
    return JsonCreative(
        title: json['title'] as String,
        subTitle: json['subTitle'] as String,
        bullets: List<String>.from(json['bullets'] as List),
        buttonText: json['buttonText']);
  }

  Map<String, dynamic> toJson() {
    return {
      'title': title,
      'subTitile': subTitle,
      'bullets': bullets,
      'buttonText': buttonText
    };
  }
}
